### ISSUE TITLE
Debugger attach to process fails with NullPointerException on WinDbg

### ISSUE THREAD

**Issue by lightspot21** [NationalSecurityAgency/ghidra#5158]
Hello there,
Attempting to attach the debugger to an already-running process using the "Attach" button of the Objects panel fails with an "attach failed" message if using the WinDbg backend.

Modifying Ghidra to print the actual exception message on failure (file: Ghidra/Debug/Debugger-agent-dbgeng/src/main/java/agent/dbgeng/model/iface1/DbgModelTargetAttacher.java) as below (lines 46-48)
```
	}).exceptionally(exc -> {
			Msg.error(this, "attach failed" + exc.getMessage());
			return null;
```
presents this on the agent window:
```
attach failed 
java.lang.NullPointerException: Cannot invoke "agent.dbgeng.manager.DbgProcess.getPid()" because "this.currentProcess" is null
```
No other modifications to source code were done.

Steps to reproduce:
1. Start a new Ghidra project
2. Open the Debugger and create a new WinDbg connection from the button on Debugger Targets. Use the default settings. 
The choice between GADP or not doesn't seem to matter.
3. The connection should complete. From the Objects panel, expand "Available" sub-tree and select a random user process.
4. Click the "Attach" button on the Objects panel, while the user process is selected.
5. The button should have no effect, and on the agent window (if using GADP) the above message will appear.

OS: Windows 10
Version: `master` as it was on March 19th 2023. Cannot test on current master right now, so not sure if it repro's on the current state of the branch.

**Comment by d-millar** [comment]
@lightspot21 Nice catch - much appreciated!  Adding the fix now.  In the event you'd prefer not to wait, the fixes are:

(1) in DbgProcessImpl::attach line 265, insert "id = DebugProcessRecord(toPid);"
(2) In DbgManageImpl::setActiveThread and setActiveProcess, wrap the first three lines of each with "if (currentThread != null) { .... }" and "if (currentProcess != null) { .... }"

I failed to adequately test the "attach" case.  (Need to check our JUnits as well, I guess.)   THANK YOU!

**Comment by ryanmkurtz** [comment]
Fixed by 33958e386da22c435f664deab2ecf4c4b647361d
